### PR TITLE
docs(api): Update get team projects api to use drf/openapi

### PR DESCRIPTION
**Pull request by schew2381**
Update GET team projects to use drf-spectacular. All that's changed is the example 200 response and permission scopes. The new response can be seen in this [github gist](https://gist.github.com/schew2381/6dcad6f6d2a6f452b0e98427df1503ab)

Before:
Screenshot 2023-06-09 at 1 57 52 PM

After:
Screenshot 2023-06-09 at 2 00 02 PM

### CODE REVIEW

File: api-docs/openapi.json [hunk 1]
```diff
@@ -93,9 +93,6 @@
     "/api/0/teams/{organization_slug}/{team_slug}/": {
       "$ref": "paths/teams/by-slug.json"
     },
-    "/api/0/teams/{organization_slug}/{team_slug}/projects/": {
-      "$ref": "paths/teams/projects.json"
-    },
```
Comment (schew2381) [inline api-docs/openapi.json:98]: Delete path so both POST and GET can use drf-spectacular

File: tests/apidocs/endpoints/events/test_project_events.py
Comment (schew2381) [inline tests/apidocs/endpoints/events/test_project_events.py:1]: Tested the GET but can be safely deleted b/c type checking is done through drf-spectacular

File: src/sentry/api/serializers/models/project.py [hunk 2]
```diff
@@ -231,6 +231,7 @@ class ProjectSerializerBaseResponse(_ProjectSerializerOptionalBaseResponse):
     firstTransactionEvent: bool
     access: List[str]
     hasAccess: bool
+    hasMinifiedStackTrace: bool
```
Comment (schew2381) [inline src/sentry/api/serializers/models/project.py:234]: `Project` model typed dict was missing this field

File: src/sentry/apidocs/examples/organization_examples.py [hunk 3]
```diff
@@ -37,11 +37,14 @@ class OrganizationExamples:
                     "hasSessions": True,
                     "hasProfiles": True,
                     "hasReplays": True,
+                    "hasMinifiedStackTrace": False,
```
Comment (schew2381) [inline src/sentry/apidocs/examples/organization_examples.py:40]: This had to be added to this OpenApiExample to comply with adding the field to the Project Model typed dict

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**